src/librustc_trans/back/write.rs

```diff
@@ -512,6 +517,13 @@ unsafe fn optimize_and_codegen(cgcx: &CodegenContext,
         None => {},
     }
 
+    // Inject a local-scoped `_Unwind_Resume` function, which redirects to `rust_eh_unwind_resume`.
+    // This must be done after optimizations, because until DwarfEHPrepare pass has been run,
+    // `_Unwind_Resume` is not referenced by any live code and would have been eliminated by opt.
```

Comment:
Wouldn’t [OptimizeNone](http://llvm.org/docs/doxygen/html/Attributes_8h_source.html#l00098) prevent it from being eliminated?